src/array.cr

```diff
@@ -577,6 +577,44 @@ class Array(T)
     indexes.map { |index| self[index] }
   end
 
+  # By using binary search, returns the first element
+  # for which the passed block returns `true`.
+  #
+  # If the block returns `false`, the finding element exists
+  # behind. If the block returns `true`, the finding element
+  # is itself or exists infront.
+  #
+  # Binary search need sorted array, so self have to be sorted.
+  #
+  # ```
+  # [2, 5, 7, 10].bsearch{ |x| x >= 4 } # => 5
+  # [2, 5, 7, 10].bsearch{ |x| x > 10 } # => nil
+  # ```
+  #
+  # Returns `nil` if the block didn't return `true` for any element.
```

Comment:
I would put this comment before the example. After the example it's less easy to spot